src/gl/texture.js

```diff
@@ -299,6 +316,7 @@ Texture.createFromObject = function (gl, textures) {
 Texture.changed = function (name, config) {
     if (Texture.textures[name]) { // cached texture
         // compare definitions
+        // TODO: fix for canvas/image/data
```

Comment:
`canvas`/`image` elements won't be stringified, which means that differences in them won't be detected. We probably need to go ahead and write (or use an existing) deep comparison method.